src/alloc/isolated_alloc.rs

```diff
@@ -255,18 +279,19 @@ impl IsolatedAlloc {
     /// SAFETY: Same as `dealloc()` with the added requirement that `layout`
     /// must ask for a size larger than the host pagesize.
     unsafe fn dealloc_huge(&mut self, ptr: *mut u8, layout: Layout) {
-        let layout = IsolatedAlloc::huge_normalized_layout(layout, self.page_size);
+        let size = self.huge_normalized_layout(layout, self.page_size);
         // Find the pointer matching in address with the one we got
         let idx = self
             .huge_ptrs
             .iter()
             .position(|pg| ptr.addr() == pg.0.addr())
```

Comment:
This will not find anything if the pointer had to be offset to ensure the requested alignment.

Please add a native-lib test that requests 4k of memory with alignment 64k. That will then likely explode here.